numba/tests/test_withlifting.py

```diff
@@ -0,0 +1,246 @@
+        the_ir = get_func_ir(func)
+        new_ir, extracted = with_lifting(
+            the_ir, self.typingctx, self.targetctx, self.flags, locals={},
+        )
+        self.assertEqual(len(extracted), expect_count)
+        cres = self.compile_ir(new_ir)
+
+        with captured_stdout() as out:
+            cres.entry_point()
+
+        self.assertEqual(out.getvalue(), expected_stdout)
+
+    def compile_ir(self, the_ir, args=(), return_type=None):
+        typingctx = self.typingctx
+        targetctx = self.targetctx
+        flags = self.flags
+        # Register the contexts in case for nested @jit or @overload calls
+        with cpu_target.nested_context(typingctx, targetctx):
+            return compile_ir(typingctx, targetctx, the_ir, args,
+                              return_type, flags, locals={})
+
+
+class TestLiftByPass(BaseTestWithLifting):
+
+    def test_lift1(self):
+        self.check_extracted_with(lift1, expect_count=1,
+                                  expected_stdout="A\nC\n")
+
+    def test_lift2(self):
+        self.check_extracted_with(lift2, expect_count=2,
+                                  expected_stdout="A 1\nD 3\n")
+
+    def test_lift3(self):
+        self.check_extracted_with(lift3, expect_count=1,
+                                  expected_stdout="A 1 100\nD 2 101\n")
+
+    def test_lift4(self):
+        self.check_extracted_with(lift4, expect_count=2,
+                                  expected_stdout="A 0\nE 11\n")
+
+    def test_lift5(self):
+        self.check_extracted_with(lift5, expect_count=0,
+                                  expected_stdout="A\n")
+
+
+class TestLiftCall(BaseTestWithLifting):
+
+    def check_same_semantic(self, func):
+        """Ensure same semantic with non-jitted code
+        """
+        jitted = njit(func)
+        with captured_stdout() as got:
+            jitted()
+
+        with captured_stdout() as expect:
+            func()
+
+        self.assertEqual(got.getvalue(), expect.getvalue())
+
+    def test_liftcall1(self):
+        self.check_extracted_with(liftcall1, expect_count=1,
+                                  expected_stdout="A 1\nB 2\n")
+        self.check_same_semantic(liftcall1)
+
+    def test_liftcall2(self):
+        self.check_extracted_with(liftcall2, expect_count=2,
+                                  expected_stdout="A 1\nB 2\nC 12\n")
+        self.check_same_semantic(liftcall2)
+
+    def test_liftcall3(self):
+        self.check_extracted_with(liftcall3, expect_count=2,
+                                  expected_stdout="A 1\nB 2\nC 47\n")
+        self.check_same_semantic(liftcall3)
+
+    def test_liftcall4(self):
+        with self.assertRaises(errors.TypingError) as raises:
+            njit(liftcall4)()
+        # Known error.  We only support one context manager per function
+        # for body that are lifted.
+        self.assertIn("re-entrant", str(raises.exception))
```

Comment:
Any view on doing something with non-existent context managers? I just accidentally forgot to import the `call_context`, but this equally applies to e.g.
```python
from numba import njit
def bar():
    with unknown_ctx:
        pass
njit(bar)()
```
which gives the same fairly spectacular traceback (as the one if you forget the import) ending with:
```
<path>/numba/numba/transforms.py in with_lifting(func_ir, typingctx, targetctx, flags, locals)
    293         _legalize_with_head(blocks[blk_start])
    294         cmkind = _get_with_contextmanager(func_ir, blocks, blk_start)
--> 295         sub = cmkind.mutate_with_body(func_ir, blocks, blk_start, blk_end,
    296                                       body_blocks, dispatcher_factory)
    297         sub_irs.append(sub)

AttributeError: Failed at nopython (nopython frontend)
'object' object has no attribute 'mutate_with_body'
```

Reply:
i think this should be fixed in this PR